Makefile

```diff
@@ -91,11 +91,11 @@ OMP_SIMD_FLAG.clang     := $(OMP_SIMD_FLAG.gcc)
 OMP_SIMD_FLAG.icc       := -qopenmp-simd
 OPT.gcc                 := -ffp-contract=fast
 OPT.clang               := $(OPT.gcc)
-CFLAGS.gcc              := -fPIC -std=c99 -Wall -Wextra -Wno-unused-parameter -MMD -MP
+CFLAGS.gcc              := -fPIC -std=c99 -Wall -Wextra -Wno-unused-parameter -Wno-unused-function -MMD -MP
 CFLAGS.clang            := $(CFLAGS.gcc)
 CFLAGS.icc              := $(CFLAGS.gcc)
 CFLAGS.XL               := -qpic -MMD
-CXXFLAGS.gcc            := -fPIC -std=c++11 -Wall -Wextra -Wno-unused-parameter -MMD -MP
+CXXFLAGS.gcc            := -fPIC -std=c++11 -Wall -Wextra -Wno-unused-parameter -Wno-unused-function -MMD -MP
```

Comment:
I'd rather drop this and be more precise about where to ignore this warning.
```c
#define CEED_UNUSED __attribute__((unused))
```

Reply:
Any preference on where the definition of `CEED_UNUSED` should go? 
Maybe `ceed-backend.h`, with `CEED_INTERN` (also uses `__attribute__`)?

Reply:
That sounds good.

Reply:
Done in a52c4238f28ab92146d7f2c90434fd06ae9b8454